src/context/context.ts

```diff
@@ -39,8 +39,8 @@ export class Context {
     this.environments = env
     this.strictVariables = renderOptions.strictVariables ?? this.opts.strictVariables
   }
-  public getRegister (key: string) {
-    return (this.registers[key] = this.registers[key] || {})
+  public getRegister (key: string, defaultValue?: any) {
```

Comment:
I don't suppose we need change `getRegister`, could you explain?

Reply:
Sorry, now I get it. You want default to `[]` in this case. Please ignore this question.

Reply:
We could leave `getRegister` alone and store the "parentloop stack" in a property of the default object, if you prefer.